sbatch_scripts/gen_sbatch.py

```diff
@@ -0,0 +1,406 @@
+
+    Returns:
+        sbatch_script (sbatch_template): SBATCH script with metadata used for running SBATCH jobs
+
+    """
+
+    sbatch_script = SBATCH_TEMPLATE.format(
+        mem=mem,
+        job_name=job_name,
+        time=time,
+        output=output,
+        error=error,
+        command=command,
+        environment=environment
+    )
+    return sbatch_script
+
+def write_sbatch(script, sbatch_script):
+    """
+    Adds SBATCH metadata to .sh script
+
+    Parameters:
+        script (.sh file): .sh file that will be turned into an SBATCH script
+        sbatch_script (string): output from the create_sbatch_script function with SBATCH metadata
+
+    Returns:
+        None.
+    """
+    with open(script, "w") as f:
+        f.write(sbatch_script)
+
+def submit_sbatch(script, logger):
+    """
+    Submits SBATCH .sh file to OSCAR.
+
+    Parameters:
+        script (SBATCH .sh): script with SBATCH metadata added after running write_sbatch
+        logger (Logger): logger from file
+    """
+    try:
+        result = subprocess.run(["sbatch", script], capture_output=True, text=True)
+        logger.info(result.stdout)
+        if result.stderr:
+            logger.error(result.stderr)
+    except Exception as e:
+        logger.error(f"Failed to submit job: {e}")
+
+def get_basic_config_from_yaml(
+    yaml_config_path: str | Path
+):
+    """
+    Load the basic configuration from a YAML file. Modified from the generate.py file
+    
+    Parameters:
+        yaml_config_path (string or Path): path to the .yaml config file for use with generate.py
+
+    Returns:
+        basic_config (dictionary?): basic configuration dictionary, taken from the config.yaml file
+    
+    """
+    basic_config = yaml.safe_load(open(yaml_config_path, "rb"))
+    return basic_config
+
+def get_environment_setup(make_env: bool, command: str):
+    """
+    Sets up environment for running generate or jaxtrain if the --make-env argument is sued
+
+    Parameters:
+        make_env (Boolean): When True, sets up environment to run genreate or jaxtrain in the SBATCH script
+        command (str): Either "generate" or "jaxtrain" depending on which script you are running
+    """
+    if make_env:
+
+        if command == "generate":
+
+            environment = textwrap.dedent("""
+            module load python
+            module load gcc
+            git clone https://github.com/lnccbrown/ssm-simulators.git
+            cd ssm-simulators
+            git checkout add-generate-to-release-0.8.3
+            uv sync""")
+
+        elif command == "jaxtrain":
+
+            environment = textwrap.dedent("""
+            module load python
+            module load gcc
+            uv add git+https://github.com/lnccbrown/lanfactory
+            cd LANfactory
+            uv sync""")
+            
+    else:
+        environment=""
+
+    return environment
+    
+def get_parameters_setup(command: str, args: dict):
```

Comment:
The `args` object is more like a namedtuple than a `dict`. Note that one doesn't use brackets to access its members.
```suggestion
def get_parameters_setup(command: str, args: argparse.Namespace):
```